### ISSUE TITLE
 inconsistency in the data types used in read and write

### ISSUE THREAD

**Issue by zhaijack** [apache/bookkeeper#750]
**FEATURE REQUEST**

1. Please describe the feature you are requesting.
there is an inconsistency in the data types used in read and write.

2. Indicate the importance of this issue to you (blocker, must-have, should-have, nice-to-have). Are you currently using any workarounds to address this issue?
should-have.

3. Provide any additional detail on your proposed use case for this feature.
In PR #738 , Ivan pointed out: 
```
 there is an inconsistency in the data types used in read and write.

WriteHandle is takes ByteBuf and ByteBuffer
ReadHandle/LedgerEntry returns ByteBuf and byte[]

This should be resolved.
```

**Comment by sijie** [comment]
marked this a blocker. we need this for the new api.

**Comment by sijie** [comment]
picking this up for unblocking 4.6.0
